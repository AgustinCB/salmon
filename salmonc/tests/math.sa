print 1+3;
print 3-1;
print 3*1;
print 1/3;