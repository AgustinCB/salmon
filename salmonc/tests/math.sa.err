Constants: [Integer(1), Integer(3), String(0)]
Instructions: [Instruction { instruction_type: Constant(0), location: 0 }, Instruction { instruction_type: Constant(1), location: 0 }, Instruction { instruction_type: Plus, location: 0 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: Minus, location: 1 }, Instruction { instruction_type: Constant(1), location: 2 }, Instruction { instruction_type: Constant(0), location: 2 }, Instruction { instruction_type: Mult, location: 2 }, Instruction { instruction_type: Constant(0), location: 3 }, Instruction { instruction_type: Constant(1), location: 3 }, Instruction { instruction_type: Div, location: 3 }]
Locations: [Location { address: 0, line: 0 }, Location { address: 0, line: 1 }, Location { address: 0, line: 2 }, Location { address: 0, line: 3 }]
