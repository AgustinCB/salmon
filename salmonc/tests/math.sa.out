0 - Integer(0)
1 - Integer(3)
2 - Integer(2)
3 - Integer(4)
